bin/inx-chronicle/src/api/stardust/indexer/routes.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use axum::{routing::get, Extension, Router};
+use chronicle::{
+    bson::{BsonExt, DocExt},
+    db::{
+        model::{inclusion_state::LedgerInclusionState, stardust::message::MessageRecord},
+        MongoDatabase,
+    },
+    stardust::{output::OutputId, payload::transaction::TransactionId},
+};
+use futures::TryStreamExt;
+use mongodb::{bson::doc, options::FindOptions};
+
+use super::{
+    extractors::{MessagesQuery, OutputsQuery},
+    responses::{MessagesForQueryResponse, OutputsForQueryResponse},
+};
+use crate::api::{
+    extractors::{Expanded, Pagination, TimeRange},
+    responses::Record,
+    stardust::{end_milestone, start_milestone},
+    ApiError, ApiResult,
+};
+
+pub fn routes() -> Router {
+    Router::new()
+        .nest("/messages", Router::new().route("/", get(messages_query)))
+        .nest("/outputs", Router::new().route("/", get(outputs_query)))
+}
+
+async fn messages_query(
+    database: Extension<MongoDatabase>,
+    query: MessagesQuery,
+    Pagination { page_size, page }: Pagination,
+    Expanded { expanded }: Expanded,
+    TimeRange {
+        start_timestamp,
+        end_timestamp,
+    }: TimeRange,
+) -> ApiResult<MessagesForQueryResponse> {
+    let MessagesQuery { tag, included: _ } = &query;
+    let start_milestone = start_milestone(&database, start_timestamp).await?;
+    let end_milestone = end_milestone(&database, end_timestamp).await?;
+
+    let mut query_doc = doc! { "milestone_index": { "$gt": start_milestone, "$lt": end_milestone } };
+    if let Some(tag) = tag.as_ref() {
+        query_doc.insert("message.payload.tag", tag);
+    }
+
+    let messages = database
+        .doc_collection::<MessageRecord>()
+        .find(
+            query_doc,
+            FindOptions::builder()
+                .skip((page_size * page) as u64)
+                .sort(doc! {"milestone_index": -1})
+                .limit(page_size as i64)
```

Comment:
what happens if this gets truncated?

Reply:
Pretty unlikely that a request with that large of a page size would succeed anyway.

Reply:
then what about using `u32`?